src/diffusers/models/embeddings.py

```diff
@@ -507,6 +626,88 @@ def forward(self, timestep, class_labels, hidden_dtype=None):
         return conditioning
 
 
+class HunyuanDiTAttentionPool(nn.Module):
```

Comment:
to-do[1] (low prior): see if we can consolidate this class with https://github.com/huggingface/diffusers/blob/bc108e15333cb0e8a092647320cbb4d70d6d0f03/src/diffusers/models/embeddings.py#L588 and https://github.com/huggingface/diffusers/blob/bc108e15333cb0e8a092647320cbb4d70d6d0f03/src/diffusers/models/unets/unet_kandinsky3.py#L483
can ask community for help